### ISSUE TITLE
Android 12 via [`15755084`](https://cs.android.com/android/_/android/platform/frameworks/base/+/157550849f0430181fa53c8e1b63112c59c6937b) and updated via [`5706277f`](https://cs.android.com/android/_/android/platform/frameworks/base/+/5706277f2d43d5be20a5510d1fb87a53c901ed52) has added a mechanism to monitor forked child processes started by apps and kills them if they consume too much CPU if app is in background.

### ISSUE THREAD

**Issue by YACINE1990** [termux/termux-app#3881]
Android 12 via [`15755084`](https://cs.android.com/android/_/android/platform/frameworks/base/+/157550849f0430181fa53c8e1b63112c59c6937b) and updated via [`5706277f`](https://cs.android.com/android/_/android/platform/frameworks/base/+/5706277f2d43d5be20a5510d1fb87a53c901ed52) has added a mechanism to monitor forked child processes started by apps and kills them if they consume too much CPU if app is in background.

>Track the child processes that are forked by app processes

>Apps could use Runtime.exec() to spawn child process and framework
will have no idea about its lifecycle. Now track those processes
whenever we find them - currently during the cpu stats sampling
they could be spotted. If it's consuming too much CPU while its
parent app process are also in the background, kill it.

>By default we allow up to 32 such processes; the process with the
worst oom adj score of their parents will be killed if there are
too many of them.

Check `PhantomProcessList` commit history [here](https://cs.android.com/android/platform/superproject/+/android-12.0.0_r4:frameworks/base/services/core/java/com/android/server/am/PhantomProcessList.java;bpv=1). The `Trimming phantom processes` log entry is from [here](https://cs.android.com/android/platform/superproject/+/android-12.0.0_r4:frameworks/base/services/core/java/com/android/server/am/PhantomProcessList.java;l=454).

Basically, not a Termux issue and users upgrading to android 12 should expect killing of background processes even from AOSP, like OEM crazy killing wasn't enough.

When you detect the killing, you can run `adb logcat -d > logcat.txt` to dump the entire logcat buffer instead of starting realtime monitoring. And are you sure you are not starting some other processes, like a sshd daemon, etc or something else, check `ps -ef`.

_Originally posted by @agnostic-apollo in https://github.com/termux/termux-app/issues/2366#issuecomment-955111262_
